extra_data/stacking.py

```diff
@@ -131,6 +131,42 @@ def stack_detector_data(train, data, axis=-3, modules=16, fillvalue=None,
     return stack
 
 
+def stack_from_xarray(xrdata, axis=-3, modules=16, fillvalue=np.nan,
+                      real_array=True):
+    
+    dtypes, shapes, empty_mods = set(), set(), set()
+    modno_arrays = {}
+    for (modno, data) in zip(xrdata.module.values, xrdata):
```

Comment:
I think this iteration will only work if modules is the first axis. We should do something like `data = xrdata.sel(module=modno)` to get the right data.